apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts

```diff
@@ -1,31 +1,55 @@
-import { HtmlSafeString, html } from '@prairielearn/html';
+import { z } from 'zod';
+
+import { HtmlSafeString, escapeHtml, html } from '@prairielearn/html';
 import { renderEjs } from '@prairielearn/html-ejs';
 
 import { Modal } from '../../components/Modal.html.js';
 
-const addSharingSetPopover = (resLocals) => {
+export const SharingSetRowSchema = z.object({
+  name: z.string(),
+  id: z.string(),
+  shared_with: z.string().array(),
+});
+type SharingSetRow = z.infer<typeof SharingSetRowSchema>;
+
+function AddSharingSetPopover({ csrfToken }: { csrfToken: string }) {
   return html`
     <form name="sharing-set-create" method="POST">
-      <input type="hidden" name="__action" value="sharing_set_create">
-      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}">
+      <input type="hidden" name="__action" value="sharing_set_create" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
 
-      <div class="form-group mb-4">
-        <p class=form-text>
-          Enter the name of the sharing set you would like to create.
-        </p>
+      <div class="form-group">
+        <label for="sharing_set_name">Sharing set name</label>
+        <input
+          class="form-control form-control-sm"
+          type="text"
+          name="sharing_set_name"
+          id="sharing_set_name"
+          required
+        />
       </div>
-      <div class=form-group>
-        <input class="form-control form-control-sm" type="text" name="sharing_set_name" required/>
-      <div>
```

Comment:
This was invalid HTML, so Prettier wasn't formatting it; that's why there are so many changes here.